examples/pytorch/gcn/gcn_ls.py

```diff
@@ -0,0 +1,281 @@
+                                                           neighbor_type='in',
+                                                           seed_nodes=np.array(seeds),
+                                                           return_seed_id=True):
+            #seed_ids = aux['seeds']
+            #print(g.in_edges(seeds, form='all'))
+            subg_src, subg_dst = subg.edges()
+            parent_nid = subg.parent_nid
+            src = parent_nid[subg_src]
+            dst = parent_nid[subg_dst]
+            parent_uv_edges_per_hop.append((src.asnumpy(), dst.asnumpy()))
+            #print((src, dst))
+            seeds = list(np.unique(src.asnumpy()))
+            induced_nodes.extend(list(parent_nid.asnumpy()))
+
+    subgraph = g.subgraph(list(np.unique(np.array(induced_nodes))))
+    #print(subgraph.parent_nid)
+    #print(parent_uv_edges_per_hop)
+    subg_uv_edges_per_hop = [(subgraph.map_to_subgraph_nid(src).asnumpy(),
+                              subgraph.map_to_subgraph_nid(dst).asnumpy())
+                             for src, dst in parent_uv_edges_per_hop]
+    #print(subg_uv_edges_per_hop)
+    return subgraph, subg_uv_edges_per_hop
+
+
+def test_sample():
+    g = generate_rand_graph(100)
+    n_layers = 3
+    seeds = [10, 20]
+    layer_size = 2
+    subgraph, subg_uv_edges_per_hop = sample_subgraph(g, seeds, n_layers, layer_size)
+
+
+class GCNLayer(gluon.Block):
+    def __init__(self,
+                 in_feats,
+                 out_feats,
+                 activation,
+                 dropout,
+                 bias=True):
+        super(GCNLayer, self).__init__()
+        with self.name_scope():
+            stdv = 1. / math.sqrt(out_feats)
+            self.weight = self.params.get('weight', shape=(in_feats, out_feats),
+                    init=mx.init.Uniform(stdv))
+            if bias:
+                self.bias = self.params.get('bias', shape=(out_feats,),
+                    init=mx.init.Uniform(stdv))
+            else:
+                self.bias = None
+        self.activation = activation
+        self.dropout = dropout
+
+    def forward(self, h, subg, subg_edges):
+        if self.dropout:
+            h = mx.nd.Dropout(h, p=self.dropout)
+        h = mx.nd.dot(h, self.weight.data(h.context))
+        # TODO: the normalization term need to be tuned.
+        # temporarilly normalized by sqrt(layer_size).
+        # for an unbiased estimator, it should be normalized by n(v)/D(v),
+        # but might be bad in practice
+        h = h / math.sqrt(3.)
+        subg.ndata['h'] = h
+        if subg_edges is not None:
+            subg.send_and_recv(subg_edges, fn.copy_src(src='h', out='m'),
+                               fn.sum(msg='m', out='h'))
+        else:
+            subg.update_all(fn.copy_src(src='h', out='m'),
+                            fn.sum(msg='m', out='h'))
+        h = subg.ndata.pop('h')
+        # same as TODO above
+        h = h / math.sqrt(3.)
```

Comment:
You can try random walk normalized Laplacian, like in https://github.com/ZiyueHuang/dgl/blob/fix_sampler1/examples/mxnet/gcn/gcn_ns.py for easier implementation,

```
def gcn_reduce(node):
    accum = mx.nd.mean(node.mailbox['m'], 1)
    return {'h': accum}
```
instead of symmetric normalized Laplacian.